### PR TITLE
Merge Ruby versions into one file

### PR DESCRIPTION
**Pull request by pikajude**
This makes adding new Ruby versions a bit easier and less copy-paste-y, in the style of PHP.

Effective changes:
- `--without-tk` and `--without-tcl` have been removed from my (unused) version of 1.8.7 as the configure script doesn't recognize them.

There is one TODO in the code regarding the new derivation for 1.8.7, which needs autoconf to build when it didn't before. I'm not sure why this is, and I can't build old 1.8.7 on my laptop at all because `missing.h` conflicts with a bunch of Libsystem stuff.

### CODE REVIEW

**Comment by vcunat** [comment]
It's possible some of the patches changes sources from which autotools generate stuff. The commit looks good to me, but I suppose it's better if some ruby user gives feedback. Trying: @psub, @cstrahan, @wkennington.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5071,23 +5071,13 @@ let
...
-  ruby_2_1_6 = callPackage ../development/interpreters/ruby/ruby-2.1.6.nix {
-    inherit (darwin) libobjc libunwind;
-  };
-  ruby_2_2_0 = callPackage ../development/interpreters/ruby/ruby-2.2.0.nix {
-    inherit (darwin) libobjc libunwind;
-  };
-  ruby_2_2_2 = callPackage ../development/interpreters/ruby/ruby-2.2.2.nix {
-    inherit (darwin) libobjc libunwind;
-  };
+  inherit (callPackage ../development/interpreters/ruby {})
```
Comment (edolstra) [inline pkgs/top-level/all-packages.nix]: This should be `callPackages` (plural).

File: pkgs/development/interpreters/ruby/default.nix [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+} @ args:
+
+let
+  op = stdenv.lib.optional;
+  ops = stdenv.lib.optionals;
+  opString = stdenv.lib.optionalString;
+  patchSet = import ./rvm-patchsets.nix { inherit fetchFromGitHub; };
+  config = import ./config.nix { inherit fetchFromSavannah; };
+
+generic = { majorVersion, minorVersion, teenyVersion, patchLevel, sha256 }: let
```
Comment (edolstra) [inline pkgs/development/interpreters/ruby/default.nix]: Indentation seems off here.

**Comment by wkennington** [comment]
Looks fine

**Pull request merged**